solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/CollectionPropertiesZkStateReader.java

```diff
@@ -72,11 +72,7 @@ public class CollectionPropertiesZkStateReader implements Closeable {
       ExecutorUtil.newMDCAwareSingleThreadExecutor(
           new SolrNamedThreadFactory("collectionPropsNotifications"));
 
-  private final ExecutorService notifications =
-      ExecutorUtil.newMDCAwareCachedThreadPool("cachecleaner");
-
-  // identify if the cleaner has already been started.
-  private final AtomicBoolean collectionPropsCacheCleanerInitialized = new AtomicBoolean(false);
+  private Thread cacheCleanerThread;
```

Comment:
Must be volatile if you wish to use double-checked locking idiom, as you are doing.  Alternatively you could always create it and instead use double-check pattern to start if not started.  Up to you.